src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1636,11 +1637,43 @@ private bool isStructRequiringStackAllocRetBuf(CORINFO_CLASS_STRUCT_* cls)
         }
 
         private CORINFO_MODULE_STRUCT_* getClassModule(CORINFO_CLASS_STRUCT_* cls)
-        { throw new NotImplementedException("getClassModule"); }
+        {
+            TypeDesc type = HandleToObject(cls).GetTypeDefinition();
```

Comment:
Can't come up with a better way to do this either. It's not great.

@BruceForstall looking at places that use this in JIT, they all seem to be:

```c
    const char* methodAssemblyName = info.compCompHnd->getAssemblyName(
        info.compCompHnd->getModuleAssembly(info.compCompHnd->getClassModule(info.compClassHnd)));
```

Could we change them to:

```c
    const char* methodAssemblyName = info.compCompHnd->getAssemblyName(
        info.compCompHnd->getModuleAssembly(info.compScopeHnd));
```

so that we can keep not implementing this API? Getting the module handle out of the class is not something we can do (this implementation will work for how JIT uses it, but the path is narrow). We can get modules from methods if needed.

Reply:
Does that work for cross-module inlining?

Reply:
`info.compScopeHnd` is the module in which the JIT resolves tokens of the method it's working on right now. So I would expect it does.

Reply:
Unfortunately, when I change the code like this, crossgen(1) crashes.

Reply:
I assume it crashes because compScopeHnd is not always a module.

Can we change this method to `getClassAssembly? It should work for all cases.

Reply:
@jkotas are you proposing there should be a new JIT-EE interface method
```
CORINFO_ASSEMBLY_HANDLE getClassAssembly(CORINFO_CLASS_HANDLE cls);
```
?

Reply:
Yes, and delete the existing `getClassModule` and `getModuleAssembly` methods since all their uses in the JIT can be replaced by `getClassAssembly`.

Reply:
btw, there was operator error: I replaced the case I cared about with the code @MichalStrehovsky gave above, but info.compScopeHnd wasn't yet defined. When I fix that, crossgen1 works. I can't verify that it's the "same" result as before, but it looks reasonable.